### PR TITLE
New: swap

### PR DESCRIPTION
**Pull request by ranneyd**
Ramda 0.29.0 introduced a [simple swap function](https://ramdajs.com/docs/#swap). Here's the typing for it. Full disclosure: I copy/pasted the code from `adjust.d.ts` since the function signatures are similar.

### CODE REVIEW

**Comment by Harris-Miller** [comment]
Nice. Just do two things for me:
* reverse the overload order (see [reduce.d.ts ](https://github.com/ramda/types/blob/develop/types/reduce.d.ts)for an example of the correct ordering)
* add tests

**Comment by ranneyd** [comment]
@Harris-Miller thanks for the quick response! On it.

**Comment by ranneyd** [comment]
@Harris-Miller could you check again now?

**Comment by ranneyd** [comment]
@Harris-Miller can you take another look?

Note: I had some lint errors. I fixed those and also updated `CONTRIBUTING.md` to call out running `npm run lint` during testing. 😄

File: types/swap.d.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  <T>(indexB: number, list: readonly T[]): T[];
+};
+
+// swap(__, indexB)
+export function swap(
+  __: Placeholder,
+  indexB: number
+): {
+  // swap(__, indexB)(indexA)(list)
+  (indexA: number): <T>(list: readonly T[]) => T[];
```
Comment (lax4mike) [inline types/swap.d.ts:19]: I'm curious about the `readonly` keyword here.  Is this just preventing the definition of `swap` from mutating the list argument?  I guess the implementation of `swap` is already written in the other repo and these definitions do not restrict that implementation.  

Does `readonly` do anything else that restricts the consumer or type inference in any way?  It probably can't hurt to have these here, but just wanted to see if I was missing something else.

**Review by lax4mike — APPROVED** [review]
Look good!

File: types/swap.d.ts (hunk 1 above)
Comment (lax4mike) [reply types/swap.d.ts:19]: ahh, that makes sense.   It does help the consumer in that this function now accepts `readonly` arrays.  Thanks for the explanation!

Comment (Harris-Miller) [reply types/swap.d.ts:19]: [playground example](https://tsplay.dev/WzxjeN)

**Review by Harris-Miller — APPROVED** [review]

**Pull request merged**